web-ui/src/context/AppContext.jsx

```diff
@@ -63,14 +74,17 @@ const AppContextProvider = props => {
   const url = `${BASE_API_URL}/csrf/cookie`;
   useEffect(() => {
     const getCsrf = async () => {
-      if (!csrf) {
+      const cookieVal = getSessionCookieValue('_csrf');
+      if (!csrf && !cookieVal) {
         const res = await fetch(url, {
           responseType: 'text',
           credentials: 'include'
         });
         if (res && res.ok) {
           dispatch({ type: SET_CSRF, payload: await res.text() });
         }
+      } else if (cookieVal) {
+        dispatch({ type: SET_CSRF, payload: cookieVal });
```

Comment:
If you checked for cookieVal before !csrf, could you remove !cookieVal from that if?

```
if (cookieVal) {
} else if (!csrf) {
}
```